types/mongoose/index.d.ts

```diff
@@ -628,7 +628,7 @@ declare module "mongoose" {
     /**
      * Defines a pre hook for the document.
      */
-    pre(method: string, parallel: boolean, fn: (next: (err?: NativeError) => void, done: () => void) => void,
+    pre(method: string, parallel: boolean, fn: (next: (err?: NativeError) => void, done: (err?: NativeError) => void) => void,
```

Comment:
Please do not use optional parameters in callbacks as discussed [here](http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html).

Reply:
Please correct me if I am wrong - but I think this is not applicable in this case.

You *definitely can* call the done()-callback without supplying a error parameter (signaling that everything was OK). It is exactly the special case mentioned in the docs: done() might be invoked with one or zero arguments.

The done()-callback is not even provided by the user but the framework itself.

Reply:
You're correct, I missed the extra level of parameter nesting. Sorry for the confusion.